static/app/views/profiling/content.tsx

```diff
@@ -388,18 +419,22 @@ function ProfilingContent({location}: ProfilingContentProps) {
                   <EnvironmentPageFilter resetParamsOnChange={CURSOR_PARAMS} />
                   <DatePageFilter resetParamsOnChange={CURSOR_PARAMS} />
                 </PageFilterBar>
-                <SearchBar
-                  searchSource="profile_landing"
-                  organization={organization}
-                  projectIds={selection.projects}
-                  query={query}
-                  onSearch={handleSearch}
-                  maxQueryLength={MAX_QUERY_LENGTH}
-                />
+                {tab === 'transactions' ? (
```

Comment:
Nit: seems like instead of littering `tab === 'transactions' and `tab === 'flamegraph` everywhere, we can just do a single top level one and render the rest of the page appropriately. There will be some code duplication but I think that's cleaner and easier to maintain as we make changes.

Reply:
Yeah, I was on the fence and didnt do it, but I agree splitting is will make it nicer to work with. I've split it to two separate components now and took the opp to extract some functionality to a separate component too